src/main/java/blacknut/ui/Blacknut.java

```diff
@@ -0,0 +1,340 @@
+//        }
+//        Task newTask = new Event(parts[0].trim(), parts[1].trim(), parts[2].trim());
+//        tasks.add(newTask);
+//        saveTasksToFile(tasks);
+//        System.out.println(" Got it. I've added this task:");
+//        System.out.println("   " + newTask);
+//        System.out.println(" Now you have " + tasks.size() + " tasks in the list.");
+//    }
+//
+//    // Method to delete a task
+//    private static void deleteTask(ArrayList<Task> tasks, String input) throws InvalidTaskNumberException {
+//        try {
+//            int index = Integer.parseInt(input.split(" ")[1]) - 1;
+//            if (index >= 0 && index < tasks.size()) {
+//                Task removedTask = tasks.remove(index);
+//                saveTasksToFile(tasks);
+//                System.out.println(" Noted. I've removed this task:");
+//                System.out.println("   " + removedTask);
+//                System.out.println(" Now you have " + tasks.size() + " tasks in the list.");
+//            } else {
+//                throw new InvalidTaskNumberException("Invalid task number. Please provide a valid number from the list.");
+//            }
+//        } catch (NumberFormatException e) {
+//            throw new InvalidTaskNumberException("Invalid command format. Please enter a valid number after 'delete'.");
+//        } catch (IndexOutOfBoundsException e) {
+//            throw new InvalidTaskNumberException("Task number out of range. Please provide a valid number from the list.");
+//        }
+//    }
+//}
+
+package blacknut.ui;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+import blacknut.ui.BlacknutExceptions.InvalidCommandException;
+import blacknut.ui.BlacknutExceptions.EmptyDescriptionException;
+import blacknut.ui.BlacknutExceptions.InvalidTaskNumberException;
+import blacknut.ui.BlacknutExceptions.IncorrectFormatException;
+import blacknut.ui.BlacknutExceptions.BlacknutException;
+
+import java.io.File;
+
+/**
+ * Represents the main Blacknut application that handles user input, processes commands,
+ * and manages tasks.
+ */
+public class Blacknut {
+    private static final String FILE_PATH = "data" + File.separator + "blacknut.txt";
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Constructs a new Blacknut application.
+     */
+    public Blacknut() {
+        ui = new Ui();
+        storage = new Storage(FILE_PATH);
+        tasks = new TaskList(storage.loadTasksFromFile());
+        parser = new Parser();
+    }
+
+    /**
+     * Runs the Blacknut application, handling user input and executing commands.
+     */
+    public void run() {
+        ui.showWelcome();
+
+        while (true) {
+            String input = ui.readInput();
+            ui.showLine();
+
+            try {
+                String command = parser.parseCommand(input);
+                switch (command) {
+                    case "bye":
```

Comment:
Remember that as per the coding standard, there is no indentation for case clauses :)